Realm/RLMSyncUser.mm

```diff
@@ -252,7 +252,7 @@ - (RLMRealm *)permissionRealmWithError:(NSError **)error {
 }
 
 - (NSURL *)authenticationServer {
-    if (!_user) {
+    if (!_user || _user->token_type() == SyncUser::TokenType::Admin) {
```

Comment:
This change isn't strictly necessary, but maybe it's good for documentation purposes?